File: WeiBoSwift/WeiBoSwift/Classes/ThirdLibs/Category/UIButton+Image.m
//
//  UIButton+Image.m
//  WeiBoSwift
//
//  Created by itogame on 2017/8/16.
//  Copyright © 2017年 itogame. All rights reserved.
//

#import "UIButton+Image.h"

@implementation UIButton (Image)

+(instancetype)buttonWithText:(NSString *)text {
    UIButton *textBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    textBtn.titleLabel.font = [UIFont systemFontOfSize:16];
    [textBtn setTitle:text forState:UIControlStateNormal];
    [textBtn setTitleColor:[UIColor grayColor] forState:UIControlStateNormal];
    [textBtn setTitleColor:[UIColor blackColor] forState:UIControlStateHighlighted];
    [textBtn setTitleColor:[UIColor blackColor] forState:UIControlStateSelected];
    [textBtn sizeToFit];
    return textBtn;
}

@end

File: WeiBoSwift/WeiBoSwift/Classes/ThirdLibs/Category/UIButton+Image.swift
//
//  UIButton+Image.swift
//  WeiBoSwift
//
//  Created by itogame on 2017/8/16.
//  Copyright © 2017年 itogame. All rights reserved.
//

import UIKit

extension UIButton {
   @objc convenience  init(imageName : String, backgroundImageName : String, titleText : String?){
        self.init()
    
    setImage(UIImage(named: imageName), for: UIControlState.normal)
    setImage(UIImage(named: imageName + "_highlighted"), for: UIControlState.highlighted)
    setImage(UIImage(named: imageName + "_selected"), for: UIControlState.selected)
    
    setBackgroundImage(UIImage(named: backgroundImageName), for: UIControlState.normal)
    setBackgroundImage(UIImage(named: backgroundImageName + "_highlighted"), for: UIControlState.highlighted)
    setBackgroundImage(UIImage(named: backgroundImageName + "_selected"), for: UIControlState.selected)
    
    setTitle(titleText, for: UIControlState.normal)
    
    sizeToFit()
    
    }
}
